python/ray/tune/trial.py

```diff
@@ -843,7 +843,8 @@ def __setstate__(self, state):
         if state["status"] == Trial.RUNNING:
             state["status"] = Trial.PENDING
         for key in self._nonjson_fields:
-            state[key] = cloudpickle.loads(hex_to_binary(state[key]))
+            if key in state:
```

Comment:
For backward compatbility I guess

Reply:
Actually this is to make the test pass first and foremost. But it doesn't hurt to keep it here

Reply:
Better is better